app/src/main/java/cloudcity/Iperf3Monitor.java

```diff
@@ -698,6 +708,10 @@ private void startIperf3Test(Context applicationContext, Iperf3RunnerData data)
                 }
             }
 
+            // Set the startTimestamp
+            testStartTimestamp = System.currentTimeMillis();
+
+            // Enqueue tasks onto the WorkManager
```

Comment:
Since when the WorkManager actually starts the worker is completely outside of our control, we don't get any benefit from this, only maybe slightly more accurate results.

As it is, the test sample I tried out was off by 3.2 seconds, since we measure teh length of the whole end-to-end (before starting, until the end of parsing) test, and not just how long data has been transmitted between the device and the iPerf3 device.

Reply:
so no, won't change, as your suggestion isn't reliable at all as well. your suggestion will keep overwriting `testStartTimestamp` as the `Worker` will be in `State.RUNNING` all the time until it either fails or succeeds.

so your commitable suggestion will actually break it and will make tests last less than one second :D

Reply:
resolving.